homeassistant/components/asterisk_mbox/__init__.py

```diff
@@ -80,7 +81,7 @@ def _discover_platform(self, component):
         )
 
     @callback
-    def handle_data(self, command, msg):
+    def handle_data(self, command, msg: dict[str, Any]) -> None:
```

Comment:
Are you sure the msg is always a `dict`? Looking at the library, I'm not completely sure tbh. Unless necessary, I would recommend to remove the type hints from this function definition.

_If it must stay, `command` should be annotated too. That is easier and should probably be `int`._

Reply:
I had typed it based on this `sorted` statement but I'm happy to remove:
```python
            self.messages = sorted(
                msg, key=lambda item: item["info"]["origtime"], reverse=True
            )
```

Reply:
Looking over the code a second time, it seems msg could be just about anything so I've removed the type hint altogether on the function.